front/app/modules/commercial/content_builder/services/contentBuilder.ts

```diff
@@ -0,0 +1,41 @@
+import { API_PATH } from 'containers/App/constants';
+import { Locale } from 'typings';
+import streams from 'utils/streams';
+
+type JsonMultiloc = {
+  [key in Locale]?: Record<string, unknown>;
+};
+
+export interface IContentBuilderLayoutData {
+  type: 'content_builder_layout';
+  id: string;
+  attributes: {
+    craftjs_jsonmultiloc: JsonMultiloc;
+    code: string;
+    enabled: boolean;
+    created_at: string;
+    updated_at: string;
+  };
+}
+
+export interface IContentBuilderLayout {
+  data: IContentBuilderLayoutData;
+}
+
+// ts-prune-ignore-next
+export function contentBuilderLayoutStream({ projectId, code }) {
+  return streams.get<IContentBuilderLayout>({
+    apiEndpoint: `${API_PATH}/projects/${projectId}/content_builder_layouts/${code}`,
+  });
+}
+
+// ts-prune-ignore-next
+export function addContentBuilderLayout(
+  { projectId, code },
+  object: IContentBuilderLayout
```

Comment:
This type is still incorrect as it includes things we will never pass in the POST request.

Please create a new type for it:

```
interface IInsightsViewObject {
  craftjs_jsonmultiloc?: JsonMultiloc;
  enabled?: boolean;
}
```

Reply:
@IvaKop Ah ok I see.

So it would be correct to have just the following? Or do they not need to be nested under the "attribute" field either?
```
export interface IContentBuilderLayoutData {
  attributes: {
    craftjs_jsonmultiloc?: JsonMultiloc;
    code?: string;
  };
}
```

Reply:
No, it would not be correct :) These are two different things - on the one case you are typing the API response to a get request and in the other - the payload for the POST request. They have different types.

Reply:
Yes I just realized that as well. Ok - adding a separate object to be used in POST requests then. Thanks!